### PR TITLE
[Woo POS] Architecture documentation, with renaming of pos files for new roles

### PR DESCRIPTION
**Pull request by joshheald**
## Description

This PR clarifies the new architecture for POS. 

To do that, I've renamed the `ItemsService` to `ItemsController`, the `ItemsProvider` is now `ItemsService`, and the Totals/Cart view models are now `ViewHelper`s.

I've also written some brief details and principles of the architecture in a README for the POS. This is far from complete, we can definitely improve on it, so treat it as a starting point!

## Testing information

There are no functional changes in this PR, a quick run through should be sufficient testing. I've done the key flows on an iPad Air running iOS 17.7.2

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Initially, we used MVVM, so you will find leftovers of that in some leaf nodes and smaller views, such as the payment messages, modals, and errors.
+
+New code should follow our Aggregate Model architecture. Some principles in brief:
+
+- State should have a single source of truth.
+- The minimum possible state for the view to do its work should be exposed.
+- The `PointOfSaleAggregateModel` coordinates globally shared state.
+- Each `View` is considered to be a view model, and can hold local state.
+- Each `Controller` handles a piece of globally shared state, often using `Services`.
+- `Services` exist to provide a stateless interface to external resources.
```
Comment (jaclync) [inline WooCommerce/Classes/POS/README.md:18]: Great to know the differences of these two.

File: WooCommerce/Classes/POS/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Services are for interacting with the network, hardware, and in future, storage. In practice, they often wrap Yosemite actions and stores in an async interface. Sometimes, they adapt existing code from the main app target to simplify its use in POS.
+
+## Future goal - Observation
+
+We intend to move to the `Observation` framework, rather than using `@Published` and `ObservableObject` on the Aggregate model.
+
+To support this, try to avoid adding subscriptions to the aggregate model's published properties. Often it's sufficient to just use a computed var. 
+
+When any published property changes, any view which has an `@EnviromentObject` var for the aggregate model will be redrawn anyway – this is part of the reason for moving to observation!
```
Comment (jaclync) [inline WooCommerce/Classes/POS/README.md]: super nit:

```suggestion
When any published property changes, any view which has an `@EnvironmentObject` var for the aggregate model will be redrawn anyway – this is part of the reason for moving to observation! 
```

File: WooCommerce/Classes/POS/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+- State should have a single source of truth.
+- The minimum possible state for the view to do its work should be exposed.
+- The `PointOfSaleAggregateModel` coordinates globally shared state.
+- Each `View` is considered to be a view model, and can hold local state.
+- Each `Controller` handles a piece of globally shared state, often using `Services`.
+- `Services` exist to provide a stateless interface to external resources.
+
+### Architectural elements
+
+#### Aggregate model
```
Comment (jaclync) [inline WooCommerce/Classes/POS/README.md]: nit: do we want to mention that it's currently Di'ed to the view hierarchy through `EnvironmentObject`?

File: WooCommerce/WooCommerceTests/POS/ViewModels/CartViewModelTests.swift [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import Testing
 @testable import WooCommerce
 
-struct CartViewModelTests {
-    let sut = CartViewModel()
+struct CartViewHelperTests {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/POS/ViewModels/CartViewModelTests.swift:4]: super nit: maybe can rename the file as well?

**Review by jaclync — APPROVED** [review]
Thanks for adding the documentation for the new architecture! LGTM :shipit:

File: WooCommerce/WooCommerceTests/POS/ViewModels/CartViewModelTests.swift (hunk 4 above)
Comment (joshheald) [reply WooCommerce/WooCommerceTests/POS/ViewModels/CartViewModelTests.swift:4]: Thanks! I knew I would miss one, but hoped I hadn't...

**Comment by staskus** [comment]
Thanks for the documentation!

**Pull request merged**